sktime/forecasting/theta_new.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file).
+"""Modular ThetaForecaster."""
+
+__author__ = ["GuzalBulatova"]
+__all__ = ["ThetaNewForecaster"]
+
+from sktime.forecasting.base._meta import _HeterogenousEnsembleForecaster
+from sktime.forecasting.compose import ColumnEnsembleForecaster
+from sktime.forecasting.compose._ensemble import _aggregate
+from sktime.forecasting.compose._pipeline import TransformedTargetForecaster
+from sktime.forecasting.exp_smoothing import ExponentialSmoothing
+from sktime.forecasting.trend import PolynomialTrendForecaster
+from sktime.transformations.series.theta import ThetaLinesTransformer
+
+
+class ThetaNewForecaster(_HeterogenousEnsembleForecaster):
+    """Modular theta method for forecasting."""
+
+    _tags = {
+        "univariate-only": False,
+        "y_inner_mtype": "pd.Series",
+        "requires-fh-in-fit": False,
+        "handles-missing-data": False,
+    }
+
+    def __init__(
+        self, theta_values=(0, 2), aggfunc="mean", weights=None, forecasters=None
+    ):
+        self.aggfunc = aggfunc
+        self.forecasters = forecasters
+        self.weights = weights
+        self.theta_values = theta_values
+        super(ThetaNewForecaster, self).__init__(forecasters=self.forecasters)
+
+    def _fit(self, y, X=None, fh=None):
+        # TODO TransformedTargetForecaster should be able to work with multivariate data
+
+        forecasters = []
```

Comment:
I have 2 small suggestions here! :)  

First - I am wondering if it could make sense to first see if `self.forecaster`'s is `None` here, and if it is one could input that into the `ColumnEnsembleForecaster` and if not could populate it based on the theta values as you do here (this suggestion is just based on what you mentioned earlier that it would be nice to allow the user more control over what forecasters are used and their values!)

Second - I know in `MultiplexForecaster` there was this convention of storing modified forecasters in `self._forecasters` rather than `self.forecasters`.  I was just thinking there might be value in doing that here as well? (ie this way you could store the forecasters you generate somewhere without modifying the passed attribute!) 

Also - some handy functionality buried within `HeterogenousEnsembleForecaster` that might be useful to you are some of the functions for taking a list of estimators and returning a list of (name, estimator) tuples - I think the signature is `self._get_estimator_tuples(estimators)`! :) So you could so something like:

``` 
def map_theta(theta):
   if theta == 0:
       return PolynomialTrendForecaster()
  return ExponentialSmoothing()
self._forecasters = self._get_estimator_tuples(list(map(map_theta, self.theta_values))
```
Of course it doesn't really do anything that different from what you are doing, and it is not blocking, just learned about it recently myself and thought it could be nice to take advantage of some of the functionality in `HeterogenousEnsembleForecaster`!

Reply:
Hi Mirae! Thanks for you feedback! 
Those are very valid points.
I think the first one should be now there. I'll also add forecasters check so that user can't pass more/less estimators than required, but if they pass None then it's populated based on theta values like you suggested :)
The second point I'll make sure to check a bit later and come back once it's done.